File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/job/JobUnlearntMessage.scala





















// Generated on 05/03/2013 10:34:45
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait JobUnlearntMessage extends MessageDeserializer {
  val messageId = 5657

  protected trait Message extends BaseMessage with Serializable {
    val messageId = JobUnlearntMessage.messageId

    val jobId: Byte

    def serialize(buf: Buffer) {
      buf.writeByte(jobId)
    }

  }

  protected def create(jobId: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val jobId = buf.readByte()

    create(jobId)
  }

}

object JobUnlearntMessage extends JobUnlearntMessage {
  case class MessageImpl(jobId: Byte) extends Message
  type Target = MessageImpl

  override def create(jobId: Byte) = MessageImpl(jobId)

  def apply(jobId: Byte) = create(jobId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/JobUnlearntMessage.java


// Generated on 05/08/2013 19:37:47
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class JobUnlearntMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5657;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte jobId;
    
    public JobUnlearntMessage() { }
    
    public JobUnlearntMessage(byte jobId) {
        this.jobId = jobId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(jobId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        jobId = buf.readByte();
        if (jobId < 0)
            throw new RuntimeException("Forbidden value on jobId = " + jobId + ", it doesn't respect the following condition : jobId < 0");
    }
    
}
